jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/applications/WindowsVariantEndpoint.java

```diff
@@ -151,4 +143,52 @@ public Response updateWindowsVariation(
         return Response.status(Response.Status.NOT_FOUND).entity("Could not find requested Variant").build();
     }
 
+    /**
+     * Get the Windows Variant for the given ID
+     *
+     * @param variantId id of {@link Variant}
+     * @return          requested {@link Variant}
+     *
+     * @statuscode 404 The requested Windows Variant resource does not exist
+     */
+    @GET
+    @Path("/{variantId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response findVariantById(@PathParam("variantId") String variantId) {
+        return doFindVariantById(variantId, WindowsVariant.class);
+    }
+
+    /**
+     * Delete the Windows Variant
+     *
+     * @param variantId id of {@link Variant}
+     *
+     * @return no content or 404
+     *
+     * @statuscode 204 The Variant successfully deleted
+     * @statuscode 404 The requested Windows Variant resource does not exist
+     */
+    @DELETE
+    @Path("/{variantId}")
+    public Response deleteVariant(@PathParam("variantId") String variantId) {
+        return this.doDeleteVariant(variantId, WindowsVariant.class);
+    }
+
+    /**
+     * Reset secret of the given Windows Variant
+     *
+     * @param variantId id of {@link Variant}
+     * @return          {@link Variant} with new secret
+     *
+     * @statuscode 200 The secret of Windows Variant reset successfully
+     * @statuscode 404 The requested Windows Variant resource does not exist
+     */
+    @PUT
+    @Path("/{variantId}/reset")
+    @Consumes(MediaType.APPLICATION_JSON)
```

Comment:
Not really part of this fix but why is the `@Consume` annotation here? It looks like this method does not receive any JSON and only uses the variantId from the path?

Reply:
No idea. Just moved this method from the base class. Let me delete it.